pvlib/modelchain.py

```diff
@@ -955,6 +960,19 @@ def no_extra_losses(self):
         self.losses = 1
         return self
 
+    def pvsyst_dc_losses(self):
+        """
+        Calculate time series of ohmic losses and apply those to the mpp power
+        output of the `dc_model` based on the pvsyst equivalent resistance
+        method. Uses a `dc_ohmic_percent` parameter in the `losses_parameters`
+        of the PVsystem.
+        """
+        Rw = self.system.pvsyst_dc_losses_eq_ohms()
+        ohmic_loss_fraction = Rw * self.dc['i_mp'] / self.dc['v_mp']
+        self.dc_ohmic_losses = ohmic_loss_fraction * self.dc['p_mp']
```

Comment:
I generally want to avoid doing any computations in `ModelChain`, so I'd put these calculations in another `pvsystem` function. I don't see a need for a corresponding `PVSystem` method.

Reply:
I was trying to model this off of the `pvwatts_losses` function, which uses a `PVSystem` method to get a parameter for the loss equation and then calculates the losses from the operating values attached to the `ModelChain`. Since this operates on `ModelChain.dc` I felt it was appropriate to have the calculation with the object it is operating on. Otherwise we will need to pass `ModelChain.dc` into a `pvsystem` function that would only be used by a `ModelChain`. That said, I'd be happy to make the change and make a new `pvsystem` function if that's the preference.